Fix MaterialVectorPostprocessor on DistributedMesh

roystgnr opened the pull request:
This catches another #1500 bug, and gets those three unit tests
working for me on at least 1 through 12 processors.

roystgnr committed:
Fix MaterialVectorPostprocessor on DistributedMesh

This catches another #1500 bug, and gets those three unit tests
working for me on at least 1 through 12 processors.

moosebuild commented:
Job `Precheck:linux-gnu` on 3e9cbd2 wanted to post the following:

Your code requires style changes.
In the top level of your repository, you can run:

`git clang-format 27fa31a91bd385d741045a16c8682b4250e340af`

Alternatively, apply the patch found [here](http://mooseframework.org/docs/PRs/8900/style.patch).
You can directly apply the patch by running:

`curl -s http://mooseframework.org/docs/PRs/8900/style.patch | git apply -v`

roystgnr commented:
I really need to build a new clang version...

roystgnr committed:
clang-format new code

I should probably merge this with the previous commit, but then I
would have no place to put commit log messages protesting the
disallowal of line wrapping.  This wrap had semantic content, even!
